### PR TITLE
[Build] Removed gRPC_PROTOBUF_PACKAGE_TYPE, supporting config only

### PR DESCRIPTION
**Pull request by veblush**
`FindProtobuf` isn't working as Protobuf began to use Abseil so gRPC is now using `CONFIG` mode for protobuf module
(Context: https://gitlab.kitware.com/cmake/cmake/-/issues/24321)

### CODE REVIEW

**Comment by veblush** [comment]
CC: @coryan

File: cmake/protobuf.cmake [hunk 1]
```diff
@@ -90,6 +86,6 @@ elseif(gRPC_PROTOBUF_PROVIDER STREQUAL "package")
         set(_gRPC_PROTOBUF_PROTOC_EXECUTABLE ${PROTOBUF_PROTOC_EXECUTABLE})
       endif()
     endif()
-    set(_gRPC_FIND_PROTOBUF "if(NOT Protobuf_FOUND AND NOT PROTOBUF_FOUND)\n  find_package(Protobuf ${gRPC_PROTOBUF_PACKAGE_TYPE})\nendif()")
+    set(_gRPC_FIND_PROTOBUF "if(NOT Protobuf_FOUND AND NOT PROTOBUF_FOUND)\n  find_package(Protobuf CONFIG)\nendif()")
```
Comment (coryan) [inline cmake/protobuf.cmake:89]: Unrelated, but this is going into `gRPCConfig.cmake`, you should use `find_dependency()` and not `find_package()` in those:

https://cmake.org/cmake/help/latest/module/CMakeFindDependencyMacro.html
https://stackoverflow.com/questions/50263280/difference-between-find-dependency-and-find-package-in-cmake

Comment (veblush) [reply cmake/protobuf.cmake]: Oh that's a good point. Thanks! Updated this and will update others too.

**Review by coryan — APPROVED** [review]

**Review by yashykt — APPROVED** [review]

File: cmake/protobuf.cmake (hunk 1 above)
Comment (carlocab) [reply cmake/protobuf.cmake]: You need to `include(CMakeFindDependencyMacro)` to use `find_dependency`. #33361

**Pull request merged**